.circleci/config.yml

```diff
@@ -0,0 +1,57 @@
+version: 2.1
+
+
+jobs:
+  build-dockerfile:
+    docker:
+      - image: cimg/base:2021.04
+    parameters:
+      image-name:
+        description: Image name
+        type: string
+      image-tag:
+        description: Image tag
+        type: string
+      target:
+        description: Dockerfile target
+        type: string
+        default: ""
+      dockerfile:
+        description: Dockerfile to use
+        type: string
+    steps:
+      - checkout
+      - setup_remote_docker:
+          version: 19.03.13
+      - run:
+          name: Build
+          command: |
+            echo "$DOCKER_PASS" | docker login -u "$DOCKER_USERNAME" --password-stdin
+            docker build -t "ethereumoptimism/<<parameters.image-name>>:<<parameters.image-tag>>" -f <<parameters.dockerfile>> <<#parameters.target>>--target <<parameters.target>><</parameters.target>> .
+            docker push "ethereumoptimism/<<parameters.image-name>>:<<parameters.image-tag>>"
+
```

Comment:
Not sure how to test this part without access, seems to be the complicated part of the PR.

Reply:
Once this is merged I'll run the job to test this out. We've used this exact code elsewhere to push images, the only issue will be setting up the Docker context which happens in the CircleCI Console.